src/components/app/data/services/subsidies/subscriptions.js

```diff
@@ -146,8 +150,13 @@ export async function activateOrAutoApplySubscriptionLicense({
   const isUserLinkedToEnterpriseCustomer = allLinkedEnterpriseCustomerUsers.some(
     (enterpriseCustomerUser) => enterpriseCustomerUser.enterpriseCustomer?.uuid === enterpriseCustomer.uuid,
   );
-  const hasActivatedSubscriptionLicense = licensesByStatus[LICENSE_STATUS.ACTIVATED].length > 0;
-  const hasRevokedSubscriptionLicense = licensesByStatus[LICENSE_STATUS.REVOKED].length > 0;
+  const isCurrentSubscriptionLicenseFilter = (subscriptionLicense) => subscriptionLicense.isCurrent;
+  const filterLicenseStatus = (licenseStatusType) => licenseStatusType.filter(
+    isCurrentSubscriptionLicenseFilter,
+  ).length > 0;
+
+  const hasActivatedSubscriptionLicense = filterLicenseStatus(licensesByStatus[LICENSE_STATUS.ACTIVATED]);
+  const hasRevokedSubscriptionLicense = filterLicenseStatus(licensesByStatus[LICENSE_STATUS.REVOKED]);
   const subscriptionLicenseToActivate = licensesByStatus[LICENSE_STATUS.ASSIGNED][0];
```

Comment:
[question] Does the assigned licenses need to have the `filterLicenseStatus` treatment, too?